libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -1112,6 +1112,48 @@ def RemoveExternalIps(self) -> Dict[str, str]:
     # Return the deleted external IP address for future use
     return external_ip_addresses
 
+  def AbandonFromMIG(self, instance_group: str) -> None:
+    """Abandons the instance from the managed instance group.
+
+    Args:
+      instance_group (str): The instance group that this instance should
+          be abondoned from.
+
+    Raises:
+      errors.ResourceDeletionError: If the request did not succeed.
+    """
+
+    def RaiseException(exception: Exception, details: str) -> None:
+      msg = ('Unable to abandon {0:s} from managed '
+             'instance group {1:s}: {2:s}').format(
+        self.name,
+        instance_group,
+        details,
+      )
+      raise errors.ResourceNotFoundError(msg, __name__) from exception
```

Comment:
maybe you can just log and leave the original exception here or add a new generic exception type "ApiOperationFailed" and wrap any exception in it.
Because if you make it a specific exception type, you should have conditional clauses on the http and runtime errors you catch below to check the error code and then decide to raise not found or resource deleted or permission denied.

Reply:
Thanks, I opted for a more generic exception type.